src/ngraph/distributed.hpp

```diff
@@ -24,6 +24,16 @@
 
 namespace ngraph
 {
+
+    enum class Reduce_t
+    {
+        sum,
+        prod,
+        min,
+        max,
+        undefined
```

Comment:
What is the purpose of `undefined` here? If we have an `undefined` value, we should check for it and reject it inside `AllReduce::validate_and_infer_types`.